src/plugins/oSnap/utils/validators.ts

```diff
@@ -145,6 +156,115 @@ export const checkIsContract = useMemoize(
     await isContractAddress(address, network)
 );
 
+export function isBool(value: string): boolean {
+  if (value === 'true' || value === 'false') {
+    return true;
+  }
+  return false;
+}
+
+export function validateInput(inputValue: string, type: InputTypes): boolean {
+  if (type === 'address') {
+    return isAddress(inputValue);
+  }
+  if (type === 'bytes') {
+    return isBytesLike(inputValue);
+  }
+  if (type === 'bytes32') {
+    return isBytesLike(inputValue);
+  }
+  if (isIntegerType(type)) {
+    return isValidInt(inputValue, type);
+  }
+  if (type === 'bool') {
+    return isBool(inputValue);
+  }
+  return true;
+}
+
+type Integer = `int${number}` | `uint${number}`;
+
+function isValidInt(value: string, type: Integer) {
+  // check if is number like
+  if (!isBigNumberish(value)) {
+    return false;
+  }
+
+  const unsigned = type.startsWith('uint');
+  const signed = type.startsWith('int');
+  if (!unsigned && !signed) {
+    throw new Error(
+      'Invalid type specified. Type must be either an unsigned integer (uint) or a signed integer (int).'
+    );
+  }
+
+  const bits = parseInt(type.slice(unsigned ? 4 : 3));
+
+  if (isNaN(bits) || bits % 8 !== 0 || bits < 8 || bits > 256) {
+    throw new Error(
+      'Invalid integer type specified. Bit size must be a multiple of 8 with a max of 256'
+    );
+  }
+
+  const number = BigNumber.from(value);
+  // range checks
+  if (unsigned) {
+    const max = BigNumber.from(2).pow(bits).sub(1);
+    return number.gte(0) && number.lte(max);
+  } else {
+    const halfRange = BigNumber.from(2).pow(bits - 1);
+    const min = halfRange.mul(-1);
+    const max = halfRange.sub(1);
+    return number.gte(min) && number.lte(max);
+  }
+}
+
+function parseArray(arrayString: string): Array<string> | undefined {
+  try {
+    const parsedValue = arrayString
+      .slice(1, -1)
+      .split(',')
+      .map(item => item.trim());
```

Comment:
safer way to parse the input text is array. safeSnap uses `JSON.parse`, but that will coerce strings like `0x000...` to `0` which is not great, we just want raw string values.
So we just strip off the square brackets and make an actual array from the contents

Reply:
im afraid this doesnt work with array of arrays, does that matter?

Reply:
Oh damn yes I think it does matter. Will have to find another solution here